File: 88-merge-sorted-array/merge-sorted-array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
let i = m + n - 1;
m--;
n--;

while (i >= 0) {
    const num1 = nums1[m];
    const num2=  nums2[n];

    if (m >= 0 && n >= 0) {
        if (num1 >= num2) {
            nums1[i] = num1;
            m--;
        } else {
            nums1[i] = num2;
            n--;
        }
    } else if (num1) {
        nums1[i] = num1;
        m--;
    } else if(num2) {
        nums1[i] = num2;
        n--;
    }

    i--;
}


};

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        last = m + n - 1
        m -= 1
        n -= 1

        while m >= 0 and n >= 0:
            num1 = nums1[m]
            num2 = nums2[n]

            if (num1 >= num2):
                nums1[last] = num1
                m -= 1
            else:
                nums1[last] = num2
                n -= 1
            last -= 1
            
        while n >= 0:
            nums1[last] = nums2[n]
            n -= 1
            last -= 1
        